src/lib.rs

```diff
@@ -126,6 +192,24 @@ fn get_identifiers(idents: &Option<CustomIdentifiers>) -> Identifiers {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, Tsify)]
+#[tsify(into_wasm_abi, from_wasm_abi)]
+enum KeyStretchingFunctionConfig {
+    #[serde(rename = "recommended")]
+    Recommended,
+    #[serde(rename = "memory-constrained")]
+    MemoryConstrained,
+    #[serde(rename = "argon2id-custom")]
+    Custom {
+        #[serde(rename = "iterations")]
+        iterations: u32,
+        #[serde(rename = "memory")]
+        memory: u32,
+        #[serde(rename = "parallelism")]
+        parallelism: u32,
+    },
```

Comment:
Maybe have a `None` option if the password is already strong or is strengthen externally...